java/org/contikios/cooja/GUI.java

```diff
@@ -1363,42 +1365,43 @@ private SwingWorker<Simulation, Object> createLoadSimWorker(Simulation.SimConfig
       addToFileHistory(configFile);
     }
 
-    final JPanel progressPanel;
-    final JDialog progressDialog;
+    ProgressMonitor progressMonitor;
     if (quick) {
-      final String progressTitle = "Loading " + (configFile == null ? "" : configFile.getAbsolutePath());
-      progressDialog = new JDialog(frame, progressTitle, Dialog.ModalityType.APPLICATION_MODAL);
-
-      progressPanel = new JPanel(new BorderLayout());
-      var progressBar = new JProgressBar(0, 100);
-      progressBar.setValue(0);
-      progressBar.setIndeterminate(true);
-
-      PROGRESS_BAR = progressBar; // Allow various parts of Cooja to show messages.
-
-      var button = new JButton("Abort");
-
-      progressPanel.add(BorderLayout.CENTER, progressBar);
-      progressPanel.add(BorderLayout.SOUTH, button);
-      progressPanel.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
-
-      progressDialog.getContentPane().add(progressPanel);
-      progressDialog.setSize(400, 200);
-
-      progressDialog.getRootPane().setDefaultButton(button);
-      progressDialog.setLocationRelativeTo(frame);
-      progressDialog.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
+      final String progressTitle = "Loading " + (configFile == null ? cooja.getSimulation().getCfg().file() : configFile.getAbsolutePath());
+      progressMonitor = new ProgressMonitor(frame, progressTitle, "", 0, 6);
+      progressMonitor.setMillisToDecideToPopup(0);
+      progressMonitor.setMillisToPopup(0);
+      progressMonitor.setProgress(0);
```

Comment:
Restore original title for ProgressMonitor if it was changed earlier.
```suggestion
      progressMonitor.setProgress(0);
      UIManager.put("ProgressMonitor.progressText", defaultTitle);
```